docs/conf.py

```diff
@@ -206,10 +206,10 @@ def generate_doxygen_xml(app):
     app : object
         The application object representing the Sphinx process.
     """
+    INPUT = os.path.join(CURR_PATH, os.path.pardir, 'include', 'LightGBM', 'c_api.h')
```

Comment:
ALL_CAPS names are used for module-level variables. Inside a function we use snake_case naming schema for variables.
```suggestion
    input = os.path.join(CURR_PATH, os.path.pardir, 'include', 'LightGBM', 'c_api.h')
```